src/pages/iou/request/step/IOURequestStepWaypoint.js

```diff
@@ -100,6 +113,65 @@ function IOURequestStepWaypoint({
         }
     }, [parsedWaypointIndex, waypointCount]);
 
+    // Construct the rectangular boundary based on user location, waypoints and direction coordinates
+    const locationBias = useMemo(() => {
+        // If there are no filled wayPoints and if user's current location cannot be retrieved,
+        // it is futile to arrive at a biased location. Let's return
+        if (filledWaypointCount === 0 && _.isEmpty(userLocation)) {
+            return null;
+        }
+
+        // Gather the longitudes and latitudes from filled waypoints.
+        const longitudes = _.filter(
+            _.map(allWaypoints, (waypoint) => {
+                if (!waypoint || lodashIsNil(waypoint.lng)) {
+                    return;
+                }
+                return waypoint.lng;
+            }),
+            (lng) => lng,
+        );
+        const latitudes = _.filter(
+            _.map(allWaypoints, (waypoint) => {
+                if (!waypoint || lodashIsNil(waypoint.lat)) {
+                    return;
+                }
+                return waypoint.lat;
+            }),
+            (lat) => lat,
+        );
+
+        // We will get direction coordinates when user is adding a stop after filling the Start and Finish waypoints.
+        // Include direction coordinates when available.
+        if (_.size(directionCoordinates) > 0) {
+            longitudes.push(..._.map(directionCoordinates, (coordinate) => coordinate[0]));
+            latitudes.push(..._.map(directionCoordinates, (coordinate) => coordinate[1]));
+        }
```

Comment:
Since the route between waypoints could have detours, I think considering direction coordinates gives us a more realistic boundary. While I too think this to be too specific, we do consider direction coordinates in arriving at the map bounds as seen [here](https://github.com/Expensify/App/blob/c01f18346f85f5120326c6aa86208c5492174d89/src/components/MapView/MapView.website.tsx#L126C1-L129C15). I think the user will naturally expect to find finer address locations within the visual map bounds. So, I think we may want to keep this. But, if you think this is way too specific, we can remove this too. Let me know what you think.

Reply:
@thienlnam 
We can either ignore direction coordinates or consider them to arrive at the rectangular boundary for location bias. Please help us decide on this.

Reply:
I don't think we need to worry about this right now - let's start with the most straightforward implementation first and then see if there's an issue

Reply:
@parasharrajat 
I have committed the changes.